prog/encodingexec_test.go

```diff
@@ -466,11 +474,17 @@ func TestSerializeForExec(t *testing.T) {
 				t.Fatalf("failed to serialize: %v", err)
 			}
 			w := new(bytes.Buffer)
-			binary.Write(w, binary.LittleEndian, test.serialized)
+			for _, v := range test.serialized {
+				tmp := make([]byte, 8)
+				*(*uint64)(unsafe.Pointer(&tmp[0])) = v
+				w.Write(tmp)
```

Comment:
All these changes... here, in prog package and in ipc package... it's not that I am opposed to every single use of unsafe... but we are adding lots of them throughout and new imports of "unsafe". Since this is now something we need to care about throughout the codebase, I am trying to consider alternatives.
For example, if we would have `binary.NativeEndian` (or `HostEndian`), it seems that it would allow to quite nice solution. Say, here, we just change s/LittleEndian/HostEndian. What do you think?
I am surprised binary package does not have it... is there something I am missing?.. whatever... we can add it ourselves.
We need to add it in some low-level enough package to avoid circular dependencies. That's probably `sys/targets` for now.
I think we can even avoid any additional overhead and indirection by doing (for each arch:
```
// little_endian.go
// +build amd64 386 arm ...
package targets
import "encdoing/binary"
var HostEndian = binary.LittleEndian
```

Reply:
This is a great idea, i like it :) I also find it abhorrent to add all those casting statements, it feels dirty. I also was surprised as i discovered that Go doesn't offer something like binary.NativeEndian.

Let me rework it then.

Reply:
Re Go, that's:
https://github.com/golang/go/issues/35398
https://github.com/golang/go/issues/36040
and:
https://github.com/golang/go/issues/37658
So it's probably not going to happen.

Reply:
A problem. targets package uses prog package already but i need HostEndian in the prog package, cycle. Any suggestion ?

Reply:
What if we make Endianness a filed of Target ? I will need it anyways later for extractFromELF ?

Reply:
1. You are right re cycle. Then I guess we need to put it into prog package.
2. I would prefer both standalone `prog.HostEndianess` and optionally as field of target. The reason for standalone is that it will allow inlining and not forcing arguments to escape.

Reply:
done